lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -430,16 +433,16 @@ migrateManually tr proxy defaultFieldValues =
                 traceWith tr $ MsgManualMigrationNotNeeded field
 
             ColumnMissing -> do
-                let orig = "checkpoint"
-                let tmp = orig <> "_tmp"
-
-                -- 1. Add genesis_hash and genesis_start to the 'wallet' table.
                 [defaults] <- runSql conn $ select ["genesis_hash", "genesis_start"] orig
                 let [PersistText genesisHash, PersistText genesisStart] = defaults
                 addColumn_ conn True (DBField WalGenesisHash) (quotes genesisHash)
                 addColumn_ conn True (DBField WalGenesisStart) (quotes genesisStart)
 
-                -- 2. Drop columns from the 'checkpoint' table
+        -- 2. Drop columns from the 'checkpoint' table
+        isFieldPresentByName conn "checkpoint" "genesis_hash" >>= \case
```

Comment:
So if the checkpoint table contains the `genesis_hash`, we do the below migration...? Won't this always be the case? Particularly after the step above?

Reply:
I got confused by that at first too. Yet, it should work fine because we are actually with two tables here:

- The wallet table
- The checkpoint table

That migration does two things:

1. It adds the genesis hash and genesis start to the _wallet_ table. 
2. It removes all the protocol parameters from the checkpoint table. 

Thus, once removed from the checkpoint table `isFieldPresentByName conn "checkpoint" "genesis_hash"` would prevent the migration from re-running again.